File: sortingAlgorithms/insertionSort/insertionsort.cpp
#include <iostream>
#include <iterator>
#include <array>

using namespace std;

int arr[] = {2,6,7,9,10,47,66,89};
int len = sizeof(arr)/sizeof(arr[0]);

void insertionsort(int arr[]);


int main(){

	insertionsort(arr);

	for(int i = 0; i < len; i++){

		cout << arr[i] << " " ;

	}

	cout << endl;

}


void insertionsort(int arr[]){

	int valuetoinsert, holepos, i;

	for(i = 0; i < len; i++){

		valuetoinsert = arr[i];

		holepos = i;

		while(holepos > 0 && arr[holepos - 1] > valuetoinsert){

			arr[holepos] = arr[holepos - 1];

			holepos--;

		}

		if(holepos != i){

			arr[holepos] = valuetoinsert;

		}

	}

}

File: sortingAlgorithms/insertionSort/insertion_sort.go
package main

import "fmt"

var arr = []int{5,3,9,7,4,2,8}

func main() {

	fmt.Println("Before Sorting: ",arr)
	insertionsort(arr)
	fmt.Println("After Sorting: ",arr)
}

func insertionsort(arr []int) {
	var valuetoinsert,holeposition int

	for i := 0; i < len(arr); i++ {
		valuetoinsert=arr[i]
		holeposition=i

		for holeposition>0 && arr[holeposition-1]>valuetoinsert {
			arr[holeposition] = arr[holeposition-1]
			holeposition--		
		}

		if holeposition!=i {
			arr[holeposition]=valuetoinsert
		}
	}
	
	
}